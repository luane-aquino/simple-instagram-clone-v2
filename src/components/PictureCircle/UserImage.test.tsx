import { render, screen } from '@testing-library/react';
import UserImage from './index';

describe('UserImage', () => {
  test('should render the component correctly', () => {
    const {asFragment} = render(<UserImage imageUrl='' name='nass' seen={false} />)
    expect(asFragment()).toMatchSnapshot()
    const button = screen.getByRole('button', {name: 'Story by nass, not seen'})
    expect(button).toBeVisible()
    const img = screen.getByRole('img', {name: 'user profile'})
    expect(img).toBeVisible()
    expect(img).toHaveAttribute('aria-disabled', 'true')
  })
})