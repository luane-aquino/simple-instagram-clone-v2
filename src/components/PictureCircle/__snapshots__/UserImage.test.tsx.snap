// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserImage should render the component correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="Story by nass, not seen"
    class="sc-bcXHqe crZyFK"
  >
    <div
      class="sc-gswNZR lnzYsQ"
    >
      <img
        alt="user profile"
        aria-disabled="true"
        class="sc-dkrFOg dHLDCZ"
        height="60"
        src=""
        width="60"
      />
    </div>
  </button>
</DocumentFragment>
`;
